app/scripts/components/analysis/define/aoi-selector.tsx

```diff
@@ -94,8 +102,17 @@ export default function AoiSelector({
     }
   }, [onAoiEvent, qsPolygon, setFeature]);
 
+  const [aoiModalRevealed, setAoIModalRevealed] = useState(false);
+
   return (
     <Fold>
+      {aoiModalRevealed && (
+        <AoIUploadModal
+          setFeature={setFeature}
+          revealed={true}
+          onCloseClick={() => setAoIModalRevealed(false)}
+        />
+      )}
```

Comment:
It would be better to use the `revealed` prop, as in doing so the modal animated in/out.

Reply:
Good point - that's what I had initially. But doing this, it's not necessary to reset state on modal re-opening (as the component is unmounted). But I guess this needs to be done anyways.

Reply:
yeah, that is true. In the past I've just used `useEffect` with revealed.

Reply:
https://github.com/NASA-IMPACT/veda-ui/pull/381/commits/8e32aaf868d2ac1548d2b36e0ff022c728c95208